main/src/ca/uwaterloo/flix/Main.scala

```diff
@@ -224,12 +225,19 @@ object Main {
           Console.println(results.output(flix.getFormatter))
         }
       case Validation.Failure(errors) =>
-        errors.sortBy(_.source.name).foreach(e => println(e.message(flix.getFormatter)))
+        createPrintableMessage(errors.sortBy(_.source.name), flix.getFormatter)
+          .foreach(println)
         println()
         println(s"Compilation failed with ${errors.length} error(s).")
         System.exit(1)
     }
 
+    def createPrintableMessage(errors: LazyList[CompilationMessage], formatter: Formatter): LazyList[String] = {
```

Comment:
- Add comment.
- Return List
- Rename to `mkMessages`.

(same for the other case)